protocol/custom-gas-token.md

```diff
@@ -0,0 +1,330 @@
+    Address B->>ERC20Converter: getERC20(recipient)  %% msg.value = native asset
+    ERC20Converter->>LiquidityController: burn() (payable)
+    LiquidityController->>Liquidity: withdraw() (payable)
+    LiquidityController-->>recipient: ERC20
+
+```
+
+To simplify the deployment process, the `L2CGTBridge` is a reserved address that contains a proxy contract, which the chain governor can point to the implementation of their conversion or bridging logic. For example, `L2CGTBridge` could point to the `ERC20Converter` implementation or to the bridge implementation if the asset is being bridged from L1 or another chain.
+
+### Native Asset Value
+
+Since native assets are pre-minted from genesis, the chain governor can decide how to assign value or give it meaning. This is done by understanding where the supply originates and how it reflects in `NativeAssetLiquidity` and `LiquidityController` management. There are at least three main cases:
+
+- The simplest is to make an existing L1 ERC20 be the native asset.
+- Introduce an ERC20 in L2 that contains utility functions (governance, DeFi); meanwhile, their native asset version serves as the medium to pay for gas.
+- Do not depend on an existing asset, emulating how some L1 currently works.
+
+The design allows for the inclusion of any desired features, such as rate limits, emission schedules, and a max. cap the supply, coupled with any bridge mechanism, etc. The architecture also doesn’t restrict the use of any token framework (e.g., OFT, xERC20, SuperchainERC20, etc.) to be coupled with the native asset eventually. Note that the `burn` function in `NativeAssetLiquidity` allows the chain governor to burn any extra supply during the chain’s deployment, enabling them to maintain only the desired supply of the native asset on the chain.
+
+Existing CGT chains using the old design can perform a hard fork to set such contracts and seed the native asset liquidity.
+
+### Wrapped Native Asset
+
+The `WETH` (now `WNA`) predeploy would stay the same as their old version, to preserve backward compatibility with existing chains.
+
+To align with the new design, the metadata, specifically the `name()` and `symbol()` functions, will now be sourced from the `LiquidityController` instead of the `SystemConfig` contract used in the previous design. These functions will return the name and symbol of the wrapped native asset, prefixed with "W" and "Wrapped".
+
+### Gas-related Components
+
+The CGT sets `Types.WithdrawalNetwork.L2` as chain type in each `FeeVaults` contracts on the constructor, without requiring contract changes. Funds are going to be sent to a desired L2 address.
+
+To accurately charge the fees for execution and data availability, chain governors may now adjust a `minBaseFee` and `operatorFee`. In the future, other parameters or mechanisms might be included to properly account for the cost for each activity, such as adding a price oracle that converts data cost expressed in ETH into the ASSET.
```

Comment:
Can you expand more on how these two parameters should be used to charge for fees? When would you adjust each and how do you know what to set it to?